### PR TITLE
[#538] add edi token expiration_date parameter

### LINKED ISSUE #538: Parameterize expiration date in API token generation
Currently there is a feature in Edi server to generate API tokens.
Tokens expiration date or expiration period is entered in secret file and there is no option to change it in runtime.
The idea is that Edi endpoint for token generation can except an optional parameter *expiration date*

### PR DESCRIPTION
**Pull request by skayerok**
This closes #538

### CODE REVIEW

**Comment by skayerok** [comment]
I think that we should probably limit the maximum expiration date, so users would not be able to set very long expiration period.
What do you think? @kirillmakhonin @mcdoker18

**Comment by kirillmakhonin** [comment]
Not sure about that. What is about adding another option: `--duration`?

**Comment by mcdoker18** [comment]
@skayerok You probably mean maximum duration, not an expiration date. I like the idea. But you should ask @aliaksandr-d  about it

**Comment by skayerok** [comment]
Yeah, I mean adding some condition to check the length of `expiration_date - current_date`.
And to set some limit on such length.

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

File: tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot [hunk 1]
```diff
@@ -226,3 +226,21 @@ Get token from EDI
     ${res} =  Shell  legionctl generate-token --edi ${HOST_PROTOCOL}://edi-${MODEL_TEST_ENCLAVE}.${HOST_BASE_DOMAIN} --model-version ${TEST_MODEL_5_VERSION} --token wrong-token
               Should not be equal   ${res.rc}  ${0}
               Should contain        ${res.stderr}  Credentials are not correct
+
+Get token from EDI with expiration date set
+    [Documentation]  Try to get token from EDI and set it`s expiration date
+    ${expiration_date} =    Get future time           ${5}  %Y-%m-%dT%H:%M:%S
```
Comment (mcdoker18) [inline tests/robot/tests/6_legionctl_commands/6.1_check_legionctl_commands.robot]: Increase offset a little bit. 
We can have a variance with kubernetes nodes.

File: legion_test/legion_test/robot/utils.py [hunk 2]
```diff
@@ -217,6 +218,32 @@ def get_component_auth_page(url, jenkins=False, token=None):
             response = requests.get(url, timeout=10)
         return {"response_code": response.status_code, "response_text": response.text}
 
+    @staticmethod
+    def ensure_component_auth_page_requires_authorization(url, jenkins=False, token=None,
```
Comment (mcdoker18) [inline legion_test/legion_test/robot/utils.py]: I think `jenkins=False` is extra parameter here

File: legion_test/legion_test/robot/utils.py [hunk 3]
```diff
@@ -303,6 +329,25 @@ def get_current_time(time_template):
...
+        :type offset: int
+        :param time_template: time template
+        :type time_template: str
+        :return: str -- time from template
+        """
+        return (datetime.datetime.utcnow() +
+                datetime.timedelta(seconds=offset)).strftime(time_template)
+
+    @staticmethod
+    def reformat_time(time_str, initial_format, target_format):
```
Comment (mcdoker18) [inline legion_test/legion_test/robot/utils.py]: I can't find an usage of this function.

Comment (skayerok) [reply legion_test/legion_test/robot/utils.py]: It is used in test `Get token from EDI and set expiration date`.
```${res_expiration_date} =  Reformat time    ${token["exp"]}    %a, %d %b %Y %H:%M:%S GMT    ${date_format}```

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Comment by skayerok** [comment]
bot, test all

**Pull request merged**